kamui/__init__.py

```diff
@@ -74,6 +76,10 @@ def unwrap_dimensional(
         raise ValueError("x must be 2D or 3D")
     psi = x.ravel()
 
+    weights = kwargs.pop('weights', None)
```

Comment:
I still think we should explicitly define the `weights` argument in the function signature and in the docstring, as I suggested in the previous review. 

`kwargs` here means the parameters for the core methods, which are edge-based. Explicitly defining and stating that `weights` should have the same shape as `x` helps people to know the it's node-based and is different. Also, in this way, `weights` will not be captured by `kwargs` and you don't have to call `pop` to get it out.

Reply:
Here is an excerpt from the data I've been working on. The data comes from a cylindrical formation, so the second axis (azimuth in the picture) is cyclical. The weights come from an additional measurement and indicate how 'sure' we are about each pixel's value.

![wrapped_and_weights](https://github.com/user-attachments/assets/2d45ac37-3c44-47f8-b4c3-f800007cdb4a)

Here's some basic comparison of unwrapping without and with weights (converted to edge weights using the 'min' method). What we care about is that the inaccuracies are limited to smaller regions/are less extreme.

![unwrapped](https://github.com/user-attachments/assets/7228c76a-1aba-446c-a2bf-9c1b414ef6b1)

When using the edgelist method, the weights seem to have less influence on the final result.

![unwrapped_edgelist](https://github.com/user-attachments/assets/f4ad78bc-100e-4ff5-a9c4-6d3f4368cd08)

Reply:
Overall, the weights give us some improvements for the most tricky parts of the data, but even without the weights `kamui` does a great job compared to our previous solutions, so thanks a lot for your work!

Reply:
That's good to know!